app/src/main/java/com/toygar/tasks/CreateTaskActivity.kt

```diff
@@ -1,41 +1,57 @@
 package com.toygar.tasks
 
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
 import android.content.Intent
 import android.os.Bundle
 import android.view.View
-import android.widget.EditText
-import android.widget.Spinner
+import android.widget.TimePicker
+import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
-import com.toygar.tasks.models.Priority
-import com.toygar.tasks.models.Task
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProvider
+import com.toygar.tasks.databinding.ActivityCreateTaskBinding
 import com.toygar.tasks.viewmodels.CreateTaskViewModel
 import java.util.*
 
 class CreateTaskActivity : AppCompatActivity() {
-    private lateinit var editTextTaskName : EditText
-    private lateinit var spinnerTaskPriority : Spinner
-    private lateinit var editTextTaskDescription : EditText
-    private lateinit var viewModel : CreateTaskViewModel
+    private val viewModel : CreateTaskViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_create_task)
-        editTextTaskName = findViewById(R.id.editTextTaskName)
-        spinnerTaskPriority = findViewById(R.id.spinnerPriority)
-        editTextTaskDescription = findViewById(R.id.editTextDescription)
-        viewModel = CreateTaskViewModel(applicationContext)
+
+        viewModel.inserted.observe(this, {
+            if (it) startActivity(Intent(this, MainActivity::class.java))
```

Comment:
burayi farkli handle etmek lazim. CreateActivity'i zaten Mainin ustune aciyorsun. Create i kapatinca Mainin gozukuyor olmasi lazim. Sen su an ust uste Main activityler aciyosun. bu iki activity arasindaki gecisleri ACitivty Resultlarla halletmek lazim.
https://stackoverflow.com/q/10407159/9613486
https://proandroiddev.com/is-onactivityresult-deprecated-in-activity-results-api-lets-deep-dive-into-it-302d5cf6edd

Reply:
Ilerde yapıldı